File: android-wrapper/app/src/main/java/org/m0skit0/android/hms/unity/ads/OnMetadataChangedListenerWrapper.java
package org.m0skit0.android.hms.unity.ads;

public class OnMetadataChangedListenerWrapper extends com.huawei.hms.ads.reward.OnMetadataChangedListener {

    private final OnMetadataChangedListener listener;

    public OnMetadataChangedListenerWrapper(final OnMetadataChangedListener listener) {
        super();
        this.listener = listener;
    }

    @Override
    public void onMetadataChanged() {
        listener.onMetadataChanged();
    }
}

File: android-wrapper/app/src/main/java/org/m0skit0/android/hms/unity/ads/OnMetadataChangedListenerWrapper.kt
package org.m0skit0.android.hms.unity.ads

class OnMetadataChangedListenerWrapper(private val listener: OnMetadataChangedListener) : com.huawei.hms.ads.reward.OnMetadataChangedListener() {

    override fun onMetadataChanged() {
        listener.onMetadataChanged()
    }
}
